src/serac/physics/tests/thermal_dynamics_patch.cpp

```diff
@@ -208,8 +208,8 @@ class LinearSolution {
     thermal.setFluxBCs(flux_function, EntireBoundary(thermal.mesh()));
 
     // volumetric source
-    auto source_function = [temp_rate_grad](auto X, auto /* time */, auto /* u */, auto /* du_dx */) {
-      return dot(X, temp_rate_grad);
+    auto source_function = [temp_rate_grad](auto position, auto /* time */, auto /* u */, auto /* du_dx */) {
```

Comment:
why change the name to position?  Is it x or X?  Aren't thermal things typically defined in terms of x?  But maybe there is no distinction here because its single physics?

Reply:
> why change the name to position?

I wanted to reserve $X$ for the vector of reference configuration coordinates of the quadrature point. Prior to this PR, that was the only information that was contained in that argument, so it was called $X$. But after this PR, that argument becomes a tuple of { $X$, $\\frac{dX}{d\xi}$ }, so I gave the tuple a new name so $X$ could still be used in the body of the qfunction.

> Is it x or X?

it refers to the reference configuration, so $X$ is clearer to me. For context, I think a lot of the initial qfunctions that were written used $x$, which was potentially confusing. So, I've been trying to go back and change them to $X$ when I touch a relevant part of the code.

> Aren't thermal things typically defined in terms of x? But maybe there is no distinction here because its single physics?

Yes-- I think physics only ever cares about the current configuration, but for the sake of implementation it's more convenient to just provide the reference configuration quantities and let the user apply whatever displacements are appropriate.